server/src/controllers/PartyController.ts

```diff
@@ -0,0 +1,306 @@
+        case "deny":
+          await this.update(
+            res,
+            address,
+            password,
+            token,
+            dataToSet => dataToSet,
+            {
+              $pull: { pendingUserlist: user }
+            }
+          );
+          break;
+        case "remove":
+          await this.update(
+            res,
+            address,
+            password,
+            token,
+            dataToSet => dataToSet,
+            {
+              $pull: { userlist: user }
+            }
+          );
+          break;
+        case "join":
+          await this.addUserToParty(req, res);
+          break;
+        default:
+          res.status(404).send({ type: "methodUnknown", method });
+      }
+    } catch (err) {
+      PartyController.sendUnknownError(res, err);
+    }
+  };
+
+  private addUserToParty = async (req: Request, res: Response) => {
+    const grailAddress = req.body.user;
+    const partyAddress = req.body.address;
+    const result = await this.grailCollection.findOne({
+      address: grailAddress
+    });
+
+    if (!result) {
+      res.status(404).send({ type: "userNotFound", grailAddress });
+      return;
+    }
+    const party = await this.partyCollection.findOne({
+      address: partyAddress
+    });
+    if (!party) {
+      res.status(404).send({ type: "partyNotFound", partyAddress });
+      return;
+    }
+    if (
+      party.userlist.indexOf(grailAddress) !== -1 ||
+      party.pendingUserlist.indexOf(grailAddress) !== -1
+    ) {
+      res.status(409).send({ type: "conflict", grailAddress });
+      return;
+    }
+    await this.update(
+      res,
+      party.address,
+      party.password,
+      party.token,
+      dataToSet => dataToSet,
+      {
+        $addToSet: { pendingUserlist: grailAddress }
+      }
+    );
+  };
+
+  private handleDbError = async (
+    address: string,
+    password: string,
+    token: string,
+    res: Response
+  ) => {
+    await this.getByAddress(address, res, existingParty => {
+      if (existingParty.password !== password) {
+        res.status(401).send({ type: "password", address });
+      } else if (existingParty.token !== token) {
+        res.status(403).send({
+          type: "token",
+          correctToken: existingParty.token,
+          specifiedToken: token,
+          address
+        });
+      } else {
+        PartyController.sendUnknownError(res, "Database error");
+      }
+    });
+  };
+
+  private update = async (
+    res: Response,
+    address: string,
+    password: string,
+    token: string,
+    modifyDataToSaveFunc: (data: Partial<IParty>) => Partial<IParty>,
+    optionalUpdates?: UpdateQuery<IParty>
+  ) => {
+    try {
+      const updateQuery: UpdateQuery<IParty> = optionalUpdates
+        ? optionalUpdates
+        : {};
+      updateQuery.$set = modifyDataToSaveFunc({
+        token: PartyController.getToken(),
+        modified: new Date()
+      });
+      updateQuery.$inc = { updateCount: 1 };
```

Comment:
updateCount can be removed (I know I said this before and if you have a reason for it, you can keep it obviously :) )

Reply:
Yes, can be removed 👍